### PR TITLE
DROOLS-3027: [DMN Designer] Differentiate zero and non zero custom data types

### PR DESCRIPTION
**Pull request by danielzhe**
drools3027

### CODE REVIEW

**Comment by danielzhe** [comment]
There's a test failing. I'll fix that and reopen the PR.

**Comment by danielzhe** [comment]
Fixed.

**Comment by danielzhe** [comment]
Jenkins execute full downstream build

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java [hunk 1]
```diff
@@ -102,6 +108,17 @@ public void setup() {
     public void setupListItems(final List<DataTypeListItem> listItems) {
         this.listItems.innerHTML = "";
         listItems.forEach(this::appendItem);
+        toggleNoListItems();
+    }
+
+    private void toggleNoListItems() {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java]: As an aside to my main comment about tests; this method isn't really _toggling_ anything merely showing the "no items view" if there are no items to show! i.e. if something toggles, then calling this method once would do X then calling it again Y then again X then again Y. In reality if I call this method twice it does the same thing twice. A minor comment. Optional if you want to change.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListViewTest.java [hunk 2]
```diff
@@ -204,6 +209,23 @@ public void testOnAddClick() {
         verify(presenter).addDataType();
     }
 
+    @Test
+    public void testHideNoCustomItemsMessageWhenThereIsCustomItem() {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListViewTest.java]: Thanks @danielzhe 

Could you please add a test for showing the `listItemsNo` (and hiding `listItems`) when there are no items?

Also, it is probably worth adding tests to cover showing/hiding `listItems` and `listItemsNo` when the `addSubItems(..)`, `addSubItem(..)` and `removeItem(..)` methods are invoked (or at least see if you can add some more assertions/verifications to existing tests to show the `showOrHideNoCustomItemsMessage()` method is invoked)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListViewTest.java [hunk 3]
```diff
@@ -204,6 +214,32 @@ public void testOnAddClick() {
...
+        verify(listItems.classList).remove(HIDDEN_CSS_CLASS);
+        verify(listItemsNo.classList).add(HIDDEN_CSS_CLASS);
+    }
+
+    @Test
+    public void testShowNoCustomItemsMessageWhenThereIsNoCustomItem() {
+
+        view.setupListItems(new ArrayList<>());
+
+        verify(listItemsNo.classList).remove(HIDDEN_CSS_CLASS);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListViewTest.java]: Should you not also `verify(listItems.classList).add(HIDDEN_CSS_CLASS)`?

Comment (danielzhe) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListViewTest.java]: Ok, commited.

**Pull request merged**